### PR TITLE
[Python Lab] Run setup and cleanup code separately from user code

### PR DESCRIPTION
**Pull request by molly-moen**
When running user code in Python Lab, I was appending setup and cleanup code to the beginning and end of their main method. This meant we had to manually adjust line numbers in error messages relating to the main method, which was a fragile operation. @snickell pointed out we could make separate calls to `runPythonAsync` for our code and the student's code, since the environment is shared between runs. This PR makes this change and removes the unnecessary code for fixing the error messages!

In the very rare (perhaps impossible?) case where our setup code hits an error, we tell the user to refresh the page. I could see this potentially happening if the user does some very unexpected messing around with system code/working directory/etc. in their code. I don't think this could occur in expected user behavior. Refreshing the page will ensure a clean environment.

Error message in this case (generated by forcing an error to be thrown):
Screenshot 2024-08-08 at 9 07 21 AM

## Links
- jira ticket: [CT-702](https://codedotorg.atlassian.net/browse/CT-702)

## Testing story
Tested locally.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/pythonlab/pyodideWebWorker.js [hunk 1]
```diff
@@ -73,6 +80,16 @@ self.onmessage = async event => {
   self.postMessage({type: 'run_complete', message: results, id});
 };
 
+async function runInternalCode(code, id) {
```
Comment (ebeastlake) [inline apps/src/pythonlab/pyodideWebWorker.js:85]: Non-blocking/not attached to this feedback: Would it be worth naming this to indicate it returns a boolean for success or failure? Off-the-top-of-head ideas: `executeCodeAndReturnStatus` or `runCodeAndCheckSuccess` or `handleCodeAndReturnStatus`...

**Review by ebeastlake — APPROVED** [review]
Nice! 🎉

File: apps/src/pythonlab/pyodideWebWorker.js (hunk 1 above)
Comment (molly-moen) [reply apps/src/pythonlab/pyodideWebWorker.js:85]: I added a clarifying comment that talks about the return value, lmk if that makes sense!

File: apps/src/codebridge/redux/consoleRedux.ts [hunk 2]
```diff
@@ -5,7 +5,13 @@ export interface CodeBridgeConsoleState {
...
 
 export interface ConsoleLog {
-  type: 'system_out' | 'system_in' | 'img' | 'system_msg' | 'error';
+  type:
+    | 'system_out'
+    | 'system_in'
+    | 'img'
+    | 'system_msg'
+    | 'error'
+    | 'system_error';
```
Comment (snickell) [inline apps/src/codebridge/redux/consoleRedux.ts:14]: totally not worth waiting for test: but could be nice to alphabetically sort these 🤷

File: apps/src/pythonlab/pyodideWorkerManager.js [hunk 3]
```diff
@@ -49,6 +45,13 @@ pyodideWorker.onmessage = event => {
       message,
     });
     return;
+  } else if (type === 'system_error') {
+    getStore().dispatch(appendSystemError(message));
+    MetricsReporter.logError({
+      type: 'PythonLabSystemCodeError',
+      message,
+    });
+    return;
```
Comment (snickell) [inline apps/src/pythonlab/pyodideWorkerManager.js:54]: 🙏

File: apps/src/pythonlab/pyodideWorkerManager.js [hunk 4]
```diff
@@ -67,11 +70,8 @@ const asyncRun = (() => {
     id = (id + 1) % Number.MAX_SAFE_INTEGER;
     return new Promise(onSuccess => {
       callbacks[id] = onSuccess;
-      let wrappedScript = applyPatches(script);
-      wrappedScript =
-        wrappedScript + deleteCachedUserModules(source, MAIN_PYTHON_FILE);
       const messageData = {
-        python: wrappedScript,
+        python: script,
```
Comment (snickell) [inline apps/src/pythonlab/pyodideWorkerManager.js:74]: NO PATCHES!!!!!

**Review by snickell — APPROVED** [review]
Very approve, beautiful diff line count:
image

File: apps/src/codebridge/redux/consoleRedux.ts (hunk 2 above)
Comment (molly-moen) [reply apps/src/codebridge/redux/consoleRedux.ts:14]: I'll try and sneak that into another pr

**Pull request merged**